### PR TITLE
Feature/swap settings main UI

### PR DESCRIPTION
**Pull request by antonijzelinskij**
#8692ty65w
image

### CODE REVIEW

**Comment by stepanLav** [comment]
Task linked: [CU-8692ty65w Android. Base layout for Swap screen](https://app.clickup.com/t/8692ty65w)

File: common/src/main/java/io/novafoundation/nova/common/utils/images/Icon.kt [hunk 1]
```diff
@@ -26,6 +28,15 @@ fun ImageView.setIcon(icon: Icon, imageLoader: ImageLoader, builder: ExtraImageR
...
 }
 
+fun ImageView.setIconOrMakeGone(icon: Icon?, imageLoader: ImageLoader, builder: ExtraImageRequestBuilding = {}) {
+    if (icon == null) {
+        this.makeGone()
+    } else {
+        this.makeVisible()
+        setIcon(icon, imageLoader, builder)
+    }
+}
```
Comment (valentunn) [inline common/src/main/java/io/novafoundation/nova/common/utils/images/Icon.kt:38]: We can use `letOrHide` exension

File: common/src/main/java/io/novafoundation/nova/common/view/shape/ShapeProvider.kt [hunk 2]
```diff
@@ -61,6 +61,15 @@ fun Context.getInputBackground() = getCornersStateDrawable(
     idleDrawable = getRoundedCornerDrawable(R.color.input_background)
 )
 
+fun Context.getInputBackgroundError() = getCornersStateDrawable(
+    focusedDrawable = getRoundedCornerDrawableFromColors(
+        fillColor = getColor(R.color.input_background),
+        strokeColor = getColor(R.color.button_background_reject),
+        strokeSizeInDp = 1f
+    ),
+    idleDrawable = getRoundedCornerDrawable(R.color.input_background)
```
Comment (valentunn) [inline common/src/main/java/io/novafoundation/nova/common/view/shape/ShapeProvider.kt:70]: Are we sure we want to hide red border when view becomes unfocused? Not sure how is that supposed to be

File: feature-swap-impl/src/main/java/io/novafoundation/nova/feature_swap_impl/presentation/main/SwapMainSettingsFragment.kt [hunk 3]
```diff
@@ -0,0 +1,75 @@
...
+
+    override fun subscribe(viewModel: SwapMainSettingsViewModel) {
+        swapMainSettingsPayInput.amountInput.bindTo(viewModel.payInput, lifecycleScope)
+        swapMainSettingsPayInput.amountInput.bindTo(viewModel.receiveInput, lifecycleScope)
+
+        viewModel.paymentTokenMaxAmount.observe { swapMainSettingsMaxAmount.text = it }
+        viewModel.paymentAsset.observe { swapMainSettingsPayInput.setModel(it) }
+        viewModel.receivingAsset.observe { swapMainSettingsReceiveInput.setModel(it) }
+        viewModel.rateDetails.observe { swapMainSettingsDetailsRate.showValue(it) }
+        viewModel.networkFee.observe { swapMainSettingsDetailsNetworkFee.showAmount(it) }
```
Comment (valentunn) [inline feature-swap-impl/src/main/java/io/novafoundation/nova/feature_swap_impl/presentation/main/SwapMainSettingsFragment.kt:68]: I suggest to use `feeLoaderMixin` instead

File: feature-swap-impl/src/main/java/io/novafoundation/nova/feature_swap_impl/presentation/main/SwapMainSettingsViewModel.kt [hunk 4]
```diff
@@ -0,0 +1,58 @@
...
+import io.novafoundation.nova.feature_swap_impl.presentation.views.SwapAssetModel
+import io.novafoundation.nova.feature_wallet_api.presentation.model.AmountModel
+import kotlinx.coroutines.flow.Flow
+import kotlinx.coroutines.flow.MutableSharedFlow
+import kotlinx.coroutines.flow.MutableStateFlow
+
+class SwapButtonState(
+    val state: ButtonState,
+    val text: String
+)
```
Comment (valentunn) [inline feature-swap-impl/src/main/java/io/novafoundation/nova/feature_swap_impl/presentation/main/SwapMainSettingsViewModel.kt:15]: We already have `DescriptiveButtonState`

**Review by valentunn — APPROVED** [review]

File: feature-swap-impl/src/main/java/io/novafoundation/nova/feature_swap_impl/presentation/main/SwapMainSettingsFragment.kt (hunk 3 above)
Comment (antonijzelinskij) [reply feature-swap-impl/src/main/java/io/novafoundation/nova/feature_swap_impl/presentation/main/SwapMainSettingsFragment.kt:68]: I'll provide it ne next prs

**Pull request merged**